client/fileutils.c

```diff
@@ -649,6 +649,78 @@ int loadFileDICTIONARY(const char *preferredName, void *data, size_t *datalen, u
     return retval;
 }
 
+uint8_t* loadFileDICTIONARY_safe(const char *preferredName, uint8_t keylen, uint16_t *keycnt) {
```

Comment:
better to return success codes as for loadFileDICTIONARY and to return the pointer to allocated space via pointer of pointer arg

Reply:
with the next pull request, when the rewrite from @iceman1001 is done.